api/v1alpha1/promotionstrategy_types.go

```diff
@@ -78,12 +81,19 @@ type Environment struct {
 	ProposedCommitStatuses []CommitStatusSelector `json:"proposedCommitStatuses"`
 }
 
-// GetAutoMerge returns the value of the AutoMerge field, defaulting to true if the field is nil.
-func (e *Environment) GetAutoMerge() bool {
-	if e.AutoMerge == nil {
-		return true
-	}
-	return *e.AutoMerge
+type AutoMergePolicy string
+
+const (
+	AutoMergePolicyDefault               AutoMergePolicy = ""
```

Comment:
Whats the point of this default why not add a default to the spec. Is this different than OnCommitStatusSuccess?

Reply:
That's probably the way to go... I think I was trying to avoid some kind of Argo CD diff weirdness with default